test/framework/test_manager.ts

```diff
@@ -0,0 +1,112 @@
+/// FIXME(jalextowle): I have a plan for making this file type safe (with useful
+/// default types). This will make writing tests significantly easier.
+import 'mocha';
+
+import { LoggingConfig, setupApiAsync, teardownApiAsync } from '../utils/deployment';
+
+import * as actions from './actions';
+import * as assertions from './assertions';
+
+/**
+ * Constructs the default `TestManager` class.
+ * @returns TestManager Returns a `TestManager` that has been given default actions
+ *          and assertions.
+ */
+export function defaultTestManager(): TestManager {
+    return new TestManager(
+        {
+            apiGetRequestAsync: actions.apiGetRequestAsync,
+        },
+        {
+            assertFieldEqualsAsync: assertions.assertFieldEqualsAsync,
+        },
+    );
+}
+
+export type ActionType = string;
+
+export type AssertionType = string;
+
+export interface ActionInfo {
```

Comment:
Maybe just `Action` and `Assertion`?

```
Action {
    name: ActionName,
    input: ...
}

Assertion {
    name: AssertionName,
    input: ...
}
```

Reply:
I like the `name` suggestion. I'll definitely add that. 

As far as the `ActionInfo` => `Action` and `AssertionInfo` => `Assertion, I'm not completely sold yet. I originally used this naming convention, but test cases really just describe the actions and assertions that must be used rather than "being" the actions and assertions. Furthermore, I think that `Action` and `Assertion` better describe the types which I ultimately named `Action` and `Assertion`. What are your thoughts on this?